vulnerabilities/pipelines/samba_importer.py

```diff
@@ -0,0 +1,335 @@
+
+            if version_match:
+                affected_packages.append(
+                    AffectedPackage(
+                        package=base_purl, fixed_version=f"{version_match.group(1)}-fixed"
+                    )
+                )
+            else:
+                affected_packages.append(
+                    AffectedPackage(package=base_purl, fixed_version="unknown")
+                )
+
+        if not cve_ids and issue_desc:
+            synthetic_id = (
+                f"SAMBA-VULN-{date_issued.replace(' ', '-') if date_issued else 'UNKNOWN'}"
+            )
+            cve_ids.append(synthetic_id)
+
+        detailed_summary = issue_desc
+
+        for cve_id in cve_ids:
+            details = self.get_advisory_details(cve_id)
+            if not details:
+                continue
+
+            if details.get("summary"):
+                detailed_summary = details["summary"]
+
+            if details.get("description"):
+                detailed_summary = f"{detailed_summary}\n\n{details['description']}"
+
+            if details.get("affected_versions"):
+                versions_details = details.get("affected_versions")
+                fixed_vers = details.get("fixed_versions", [])
+
+                for pkg in self.extract_affected_packages_from_detail(versions_details, fixed_vers):
+                    affected_packages.append(pkg)
+
+            if details.get("mitigation"):
+                detailed_summary = f"{detailed_summary}\n\nMitigation: {details['mitigation']}"
+
+        references = []
+        for link in cells[5].find_all("a"):
+            announcement_text = link.get_text().strip()
+            announcement_url = link.get("href")
+            if announcement_url:
+                if not announcement_url.startswith(("http://", "https://")):
+                    announcement_url = f"https://www.samba.org{announcement_url}"
+
+                reference_id = None
+                for cve_id in cve_ids:
+                    if cve_id in announcement_url:
+                        reference_id = cve_id
+                        break
+
+                if not reference_id:
+                    reference_id = announcement_text
+
+                references.append(Reference(url=announcement_url, reference_id=reference_id))
+
+        for patch_url in patch_links:
+            patch_filename = patch_url.split("/")[-1]
+            references.append(Reference(url=patch_url, reference_id=f"Patch: {patch_filename}"))
+
+        return AdvisoryData(
+            aliases=cve_ids,
+            summary=build_description(
+                summary=detailed_summary, description=f"Affected versions: {affected_releases}"
+            ),
+            references=references,
+            affected_packages=affected_packages,
+            date_published=date_published,
+            url="https://www.samba.org/samba/history/security.html",
```

Comment:
Shouldn't the url be the individual announcement like https://www.samba.org/samba/security/CVE-2023-3961.html?